app/controllers/concerns/foreman/controller/registration_commands.rb

```diff
@@ -33,7 +44,11 @@ def command_headers
     }
 
     if registration_params['jwt_expiration'].present?
-      jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i if registration_params['jwt_expiration'] != 'unlimited'
+      if expiration_valid?
+        jwt_args[:expiration] = registration_params['jwt_expiration'].to_i.hours.to_i unless expiration_unlimited?
```

Comment:
Can we extract the `registration_params['jwt_expiration']` into a method, to make the `jwt_expiration` constant appear once? 
Something like this:

``` ruby
def jwt_expiration_param
  # use require if the parameter is mandatory or permit if it's optional.
  param = registration_params.require('jwt_expiration') || 4
  @jwt_expiration_param ||= begin
    if param == 'unlimited'
      0
    else
      Integer(param)
    end
end
```
Then the `expiration_unlimited` would not have to handle with string values at all:
``` ruby
def def expiration_unlimited?
  jwt_expiration_param == 0
end
```

Reply:
Thanks Shim. This was more readable. 
A few points:

- permit method returns the value as a Parameter object. So I'm unable to use to_i or Integer on it
-  I have used `param.to_i` instead of `Integer(param)` because it would break the code if we input a decimal value from the UI (which currently is converted to greatest integer with the to_i method). Also integer validation is automatically happening for hammer and api [here](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/registration_commands_controller.rb#L18) with the [number validator](https://github.com/Apipie/apipie-rails?tab=readme-ov-file#numbervalidator)

Reply:
Sadly, we do not enable Apipie validations in Foreman:
https://github.com/theforeman/foreman/blob/a1142ab9ef500e3206be1ff8a390431bc1b32de3/config/initializers/apipie.rb#L13-L14

Some clients (e.g. FAM) still use the values in the APIDoc to enforce validation, but that's up to the clients.